rust/src/io/exec/limit.rs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use arrow_array::RecordBatch;
+use futures::stream::{Stream, TryStreamExt};
+use tokio::sync::mpsc::{self, Receiver};
+use tokio::task::JoinHandle;
+
+use super::{ExecNode, NodeType};
+use crate::{Error, Result};
+
+/// Dataset Scan Node.
+pub(crate) struct Limit {
+    rx: Receiver<Result<RecordBatch>>,
+    _io_thread: JoinHandle<()>,
+}
+
+impl Limit {
+    /// Create a new execution node to handle limit offset.
+    pub fn new(
+        child: impl ExecNode + Unpin + Send + 'static,
+        limit: Option<i64>,
```

Comment:
should `limit` not a Option value

Reply:
offline discussion: postgres supports offset without limit